src/audio/google_rtc_audio_processing.c

```diff
@@ -347,16 +347,16 @@ static int google_rtc_audio_processing_copy(struct comp_dev *dev)
 	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
 
 	struct comp_copy_limits cl;
-	int frame;
-	uint32_t aec_reference_buff_frag;
-	uint32_t raw_microphone_buff_frag;
-	uint32_t output_buff_frag;
-	int16_t *src;
-	int16_t *dst;
+	int16_t *src = cd->raw_microphone->stream.r_ptr;
+	int16_t *dst = cd->output->stream.w_ptr;
+	int16_t *ref = cd->aec_reference->stream.r_ptr;
 	uint32_t num_aec_reference_frames;
 	uint32_t num_aec_reference_bytes;
 	int channel;
+	int remain;
```

Comment:
Can you rename it to `num_sample_remaining` ?

Reply:
I was rethinking it, and maybe we should stick to frames and handle the number of samples in the inner loop. The reason is that we can have different number of channel for:
  - aec reference
  - capture input
  - capture output
But the number of frame should match.

Reply:
Sure, I'll rename if no issues with too long lines. Yep, I kept the possibility for different number of channels in ref, src, and snk.
 
The samples counting is more efficient so I used it in the reference loop. The frames without wrap function is using division that is slow on xtensa.